src/main/java/org/threadly/util/debug/ControlledThreadProfiler.java

```diff
@@ -102,11 +102,30 @@ public boolean removedProfiledThread(Thread t) {
    * @return count of tracked threads.
    */
   public int getProfiledThreadCount() {
-    return profiledThreads.size();
+    return controledThreadStore.profiledThreads.size();
   }
   
-  @Override
-  protected Iterator<Thread> getProfileThreadsIterator() {
-    return profiledThreads.iterator();
+  /**
+   * <p>Extending class of {@link ProfileStorage} this overrides 
+   * {@link #getProfileThreadsIterator()}.  It controls it so that not all VM threads are returned 
+   * in the iterator, and instead it only iterates over the threads which are stored 
+   * internally.</p>
+   * 
+   * @author jent
+   * @since 3.5.0
+   */
+  protected static class ControlledThreadProfileStorage extends ProfileStorage {
```

Comment:
Here is our extending implementation of "ProfilerStorage" so that we can control what threads will be seen by the thread iterator.  Previously we did that by overriding "getProfileThreadsIterator" in the parent class, but that is no longer possible, we must override it here.